test/NuGet.Clients.Tests/NuGet.VisualStudio.Common.Test/NuGetFeedbackDiagnosticFileProviderTests.cs

```diff
@@ -1,27 +1,92 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
 using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Sdk.TestFramework;
+using Moq;
+using NuGet.PackageManagement;
+using NuGet.ProjectManagement;
 using Xunit;
 
 namespace NuGet.VisualStudio.Common.Test
 {
+    [Collection(MockedVS.Collection)]
     public class NuGetFeedbackDiagnosticFileProviderTests
     {
-        [Fact]
-        public void GetFilesTest()
+        private NuGetFeedbackDiagnosticFileProvider _target;
+        private Mock<ISolutionManager> _solutionManager;
+
+        public NuGetFeedbackDiagnosticFileProviderTests(GlobalServiceProvider globalServiceProvider)
         {
-            // Arrange
-            var provider = new NuGetFeedbackDiagnosticFileProvider();
+            // NuGetFeedbackGiagnosticFileProvider uses ThreadHelper.JoinableTaskFactory, so MockedVS is needed
+            globalServiceProvider.Reset();
+
+            _solutionManager = new Mock<ISolutionManager>();
+            _solutionManager.Setup(sm => sm.GetNuGetProjectsAsync())
+                .Returns(Task.FromResult<IEnumerable<NuGetProject>>(Array.Empty<NuGetProject>())); // empty or no solution
+
+            _target = new NuGetFeedbackDiagnosticFileProvider();
+            _target.SolutionManager = _solutionManager.Object;
+        }
 
+        [Fact]
+        public void EnsureSingleFileWithFullPathIsReturned()
+        {
+            // Arranged in constructor
             // Act
-            var files = provider.GetFiles();
+            IReadOnlyCollection<string> files = _target.GetFiles();
 
-            // Assert
-            foreach (var file in files)
+            try
             {
+                // Assert
+                // As per feedback team's docs, multiple files should be saved in a zip, so our class should only
+                // return that 1 filename.I guess the interface // returning a collection was a design mistake that
```

Comment:
nit: space after `.`. I'm not sure why `//` here either, unless it was supposed to be a new line?
```suggestion
                // return that 1 filename. Returning a collection appears to be a design mistake that
```

Reply:
oops. I was editing comments, and obviously forgot to fix/finish this one.